cli/command/system/info.go

```diff
@@ -103,7 +103,7 @@ func prettyPrintInfo(dockerCli *command.DockerCli, info types.Info) error {
 			fmt.Fprintf(dockerCli.Out(), " Error: %v\n", info.Swarm.Error)
 		}
 		fmt.Fprintf(dockerCli.Out(), " Is Manager: %v\n", info.Swarm.ControlAvailable)
-		if info.Swarm.ControlAvailable {
+		if info.Swarm.ControlAvailable && info.Swarm.Error == "" {
```

Comment:
Thanks, that is reasonable, updated within this PR: if LocalNodeStateError, do not display more swarm info. PTAL.